src/eos-updater-poll-lan.c

```diff
@@ -0,0 +1,920 @@
+                                &extensions,
+                                &local_error))
+        {
+          message ("Failed to fetch latest commit from %s: %s",
+                   url_override,
+                   local_error->message);
+          continue;
+        }
+
+      if (!is_checksum_an_update (repo, checksum, &commit, &local_error))
+        {
+          message ("Commit %s from %s is not an update",
+                   checksum,
+                   url_override);
+          continue;
+        }
+
+      timestamp = ostree_commit_get_timestamp (commit);
+      if (latest_checksum != NULL)
+        {
+          if (timestamp < latest_timestamp)
+            continue;
+
+          if (timestamp == latest_timestamp && g_strcmp0(checksum, latest_checksum) == 0)
+            {
+              g_ptr_array_add (swbfs_with_latest_commit, g_object_ref (swbf));
+              g_ptr_array_add (urls, g_steal_pointer (&url_override));
+            }
+          else if (timestamp > latest_timestamp && g_strcmp0(checksum, latest_checksum) != 0)
+            {
+              g_clear_pointer (&latest_checksum, g_free);
+              g_clear_pointer (&latest_commit, g_variant_unref);
+              latest_timestamp = 0;
+              g_ptr_array_set_size (swbfs_with_latest_commit, 0);
+              g_ptr_array_set_size (urls, 0);
+              g_clear_object (&latest_extensions);
+            }
+        }
+      if (latest_checksum == NULL)
+        {
+          latest_checksum = g_steal_pointer (&checksum);
+          latest_commit = g_steal_pointer (&commit);
+          latest_timestamp = timestamp;
+          g_ptr_array_add (swbfs_with_latest_commit, g_object_ref (swbf));
+          g_ptr_array_add (urls, g_steal_pointer (&url_override));
+          latest_extensions = g_steal_pointer (&extensions);
+        }
+    }
+
+  if (latest_checksum != NULL)
+    {
+      g_set_object (&latest_extensions->branch_file, newest);
+      *out_info = eos_update_info_new (latest_checksum,
+                                       latest_commit,
+                                       refspec,
+                                       original_refspec,
+                                       (gchar **)g_ptr_array_free (g_steal_pointer (&urls),
+                                                                   FALSE),
+                                       latest_extensions);
+    }
+  *out_metrics = g_steal_pointer (&metrics);
+
+  return TRUE;
+}
+
+static gboolean
+check_lan_updates (LanData *lan_data,
+                   GPtrArray *found_services)
+{
+  g_autoptr(GPtrArray) valid_services = NULL;
+  g_autoptr(EosBranchFile) branch_file = NULL;
+
+  if (!filter_services (lan_data,
+                        found_services,
+                        &valid_services,
+                        &lan_data->error))
+    {
+      message ("Failed to filter services: %s",
+               lan_data->error->message);
+      return TRUE;
```

Comment:
I added a small comment about the return value of this function. The returned value is telling whether we should leave the main loop after this function or not. `TRUE` - do the `g_main_loop_quit` call, `FALSE` - no.

Currently we only return `TRUE`, but maybe we will want to refine the fetching process in future, so it may become asynchronous. In that case we would return FALSE when some asynchronous actions were started.